### PR TITLE
[BACK-70] Refactoring and fixes of dbUpdater

### PR DESCRIPTION
**Pull request by thomas-woehrle**
### Description of the issue
- refactor the files dbUpdater.ts, processRepo.ts and dataAggregation.ts
- reorganize the dbUpdater functionality

### How I implemented
- Redistributed functions that reside in dbUpdater, processRepo and dataAggregation, because they did not really belong where they were
- renamed `processRepo.ts` to `updateProject.ts` and `dataAggregation.ts` to `supabaseUtils.ts`
- introduced new file `utils.ts` 
- adjusted the corresponding types files names
- introduced new supabaseUtils functions to enable the workflow as needed
- introduced getProjectGitHubStats und getProjectStarHistory to make the projectCreation completely modular

### Other
**Branched from BACK-58.** 
See BACK-58 PR description for that part of the changes. Converted BACK-58 to draft, because had no reviews yet anyway

### CODE REVIEW

File: packages/graphql_server/src/api/openAIApi.ts [hunk 1]
```diff
@@ -60,6 +60,10 @@ async function getELI5FromReadMe(readMe: string) {
  */
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 async function getHackernewsSentiment(comments: string) {
+  // limit length of name to 5000 characters, because it might cause problems with GPT otherwise
+  if (comments.length > 5000) {
```
Comment (ArnoClaude) [inline packages/graphql_server/src/api/openAIApi.ts:64]: `5000` should probably be a `const` called `GPT_MESSAGE_LIMIT` or something instead of a magic number :)

**Review by ArnoClaude — CHANGES_REQUESTED** [review]

File: packages/graphql_server/src/api/openAIApi.ts [hunk 2]
```diff
@@ -60,6 +60,10 @@ async function getELI5FromReadMe(readMe: string) {
  */
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (kigadye) [inline packages/graphql_server/src/api/openAIApi.ts:61]:
```suggestion
```

File: packages/graphql_server/src/dbUpdater.ts [hunk 3]
```diff
@@ -1,152 +1,116 @@
...
+  getPersonID,
+  purgeTrendingState,
+  repoIsAlreadyInDB,
+  deleteNotTrendingAndNotBookmarkedProjects,
+  getTrendingAndBookmarkedProjects
+} from './supabaseUtils'
+import { fetchTrendingRepos } from './scraping/githubScraping'
+/* 
+Types: 
+*/
```
Comment (kigadye) [inline packages/graphql_server/src/dbUpdater.ts:18]:
```suggestion
```

File: packages/graphql_server/src/dbUpdater.ts [hunk 4]
```diff
@@ -1,152 +1,116 @@
...
+import { TrendingState } from '../types/updateProject'
 
 /**
- * Updates the database with the current trending repositories.
- * Also Deletes or updates old projects.
+ * This is the function could be called daily to update the db.
+ * It updates old projecst and inserts new trending ones.
```
Comment (kigadye) [inline packages/graphql_server/src/dbUpdater.ts:23]:
```suggestion
 * Updates old projects and inserts new trending ones.
```

File: packages/graphql_server/src/dbUpdater.ts [hunk 5]
```diff
@@ -1,152 +1,116 @@
...
   for (let i = 0; i < repos.length / 2; i++) {
     const owner = repos[2 * i]
     const name = repos[2 * i + 1]
+    console.log(
+      '###################### Processing',
+      name,
+      'owned by',
+      owner,
+      '######################'
+    )
```
Comment (kigadye) [inline packages/graphql_server/src/dbUpdater.ts:74]:
```suggestion
```

File: packages/graphql_server/src/dbUpdater.ts [hunk 6]
```diff
@@ -1,152 +1,116 @@
...
+      return false
     } else {
-      const { data: owning_person } = await supabase
-        .from('associated_person')
-        .select('*')
-        .eq('id', repo.owning_person)
-
-      // the owner has the same name -> the repo is already in the database
-      if (owning_person?.[0]?.login === owner) return true
+      console.log('inserted project', repoName, 'owned by', owner)
```
Comment (kigadye) [inline packages/graphql_server/src/dbUpdater.ts:110]:
```suggestion
```

File: packages/graphql_server/src/dbUpdater.ts (hunk 5 above)
Comment (kigadye) [reply packages/graphql_server/src/dbUpdater.ts:74]: too many info logs on prod server will make it very hard to debug errors

File: packages/graphql_server/src/utils.ts
Comment (kigadye) [inline packages/graphql_server/src/utils.ts:1]: create utils folder and put each function into a separate file

File: packages/graphql_server/src/supabaseUtils.ts
Comment (kigadye) [inline packages/graphql_server/src/supabaseUtils.ts:1]: this file is way too big, split up functions into individual files

File: packages/graphql_server/src/supabaseUtils.ts [hunk 7]
```diff
@@ -0,0 +1,492 @@
+import supabase from './supabase'
+import { getOrganizationInfo, getUserInfo } from './api/githubApi'
+/*
+Types:
+*/
```
Comment (kigadye) [inline packages/graphql_server/src/supabaseUtils.ts:5]:
```suggestion
```

File: packages/graphql_server/src/test/dbUpdater.test.ts [hunk 8]
```diff
@@ -1,7 +1,8 @@
-import { dbUpdater } from '../dbUpdater'
+import { dailyDbUpdater } from '../dbUpdater'
 
 try {
-  void dbUpdater()
+  void dailyDbUpdater(false)
 } catch (error) {
   console.log('error occured')
```
Comment (kigadye) [inline packages/graphql_server/src/test/dbUpdater.test.ts:6]:
```suggestion
```
console.error already implies that it's an error

Comment (kigadye) [inline packages/graphql_server/src/test/dbUpdater.test.ts:6]:
```suggestion
```
console.error already implies that it's an error

File: packages/graphql_server/src/updateProject.ts [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+ * Updates the eli5 of a repo
+ * @param {string} name - The name of the repo.
+ * @param {string} owner - The name of the owner of the repo.
+ */
+const updateProjectELI5 = async (name: string, owner: string) => {
+  try {
+    const readMe = (await fetchRepositoryReadme(owner, name)).slice(0, 2500)
+    const description = await getELI5FromReadMe(readMe)
+    const updated = await updateSupabaseProject(name, owner, { eli5: description })
+    updated && console.log('updated eli5 of ', name, 'owned by', owner)
```
Comment (kigadye) [inline packages/graphql_server/src/updateProject.ts:74]:
```suggestion
```

File: packages/graphql_server/src/updateProject.ts [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+/**
+ * Updates the github stats of a repo.
+ * Not the star-history though.
+ * @param {string} repoName - The name of the repo.
+ * @param {string} owner - The name of the owner of the repo.
+ */
+const updateProjectGithubStats = async (name: string, owner: string) => {
+  const githubStats: GitHubInfo | null = await getGithubData(name, owner)
+  if (!githubStats) {
+    console.log('Could not get github stats for ', name, 'owned by', owner)
```
Comment (kigadye) [inline packages/graphql_server/src/updateProject.ts:92]:
```suggestion
    console.error('Could not get github stats for ', name, 'owned by', owner)
```

File: packages/graphql_server/src/updateProject.ts [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+    return
+  }
+  const updated = await updateSupabaseProject(name, owner, formatGithubStats(githubStats))
+
+  if (!updated) {
+    console.log('Could not update github stats for ', name, 'owned by', owner)
+  } else {
+    console.log('Updated github stats for ', name, 'owned by', owner)
+  }
```
Comment (kigadye) [inline packages/graphql_server/src/updateProject.ts:101]:
```suggestion
  if (!updated) {
    console.error('Could not update github stats for ', name, 'owned by', owner)
  }
```

**Review by kigadye — APPROVED** [review]
Didn't find any bugs, code quality is really good. I think you could refactor the codebase even further though, there's still a lot of huge files that can be split up and a lot of info logs that are not really necessary. I've seen some projects that used console.info for info logs and then configured the logger in a way that on prod only errors are logged but idk if that comes out of the box with node or requires a third party logger.

**Review by acreturs — APPROVED** [review]
Could not test the functionality since there is currently a problem on my machine. But the code looks well structured and according to the guidelines
